### DISCUSSION TITLE
Install Spacy for Python-3.10.4 on Mac M1 (Error)

### DISCUSSION THREAD

**Discussion by alankabisov** [explosion/spaCy#11478] [Help: Installation] [answered]
Hi there!

I try to Install Spacy for Python-3.10.4 on Mac M1 and get the following error when I run `!python -m spacy download en`:

> RuntimeError: module compiled against API version 0xf but this version of numpy is 0xe
> RuntimeError: module compiled against API version 0xf but this version of numpy is 0xe
> ImportError: numpy.core._multiarray_umath failed to import
> ImportError: numpy.core.umath failed to import
> Traceback (most recent call last):
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/runpy.py", line 187, in _run_module_as_main
>     mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/runpy.py", line 146, in _get_module_details
>     return _get_module_details(pkg_main_name, error)
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/runpy.py", line 110, in _get_module_details
>     __import__(pkg_name)
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/spacy/__init__.py", line 11, in <module>
>     from thinc.api import prefer_gpu, require_gpu, require_cpu  # noqa: F401
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/thinc/api.py", line 2, in <module>
>     from .initializers import normal_init, uniform_init, glorot_uniform_init, zero_init
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/thinc/initializers.py", line 4, in <module>
>     from .backends import Ops
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/thinc/backends/__init__.py", line 7, in <module>
>     from .ops import Ops
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/thinc/backends/ops.py", line 13, in <module>
>     from ..util import get_array_module, is_xp_array, to_numpy
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/thinc/util.py", line 47, in <module>
>     import tensorflow.experimental.dlpack
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/tensorflow/__init__.py", line 37, in <module>
>     from tensorflow.python.tools import module_util as _module_util
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/tensorflow/python/__init__.py", line 42, in <module>
>     from tensorflow.python import data
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/tensorflow/python/data/__init__.py", line 21, in <module>
>     from tensorflow.python.data import experimental
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/tensorflow/python/data/experimental/__init__.py", line 95, in <module>
>     from tensorflow.python.data.experimental import service
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/tensorflow/python/data/experimental/service/__init__.py", line 387, in <module>
>     from tensorflow.python.data.experimental.ops.data_service_ops import distribute
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/tensorflow/python/data/experimental/ops/data_service_ops.py", line 22, in <module>
>     from tensorflow.python.data.experimental.ops import compression_ops
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/tensorflow/python/data/experimental/ops/compression_ops.py", line 16, in <module>
>     from tensorflow.python.data.util import structure
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/tensorflow/python/data/util/structure.py", line 22, in <module>
>     from tensorflow.python.data.util import nest
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/tensorflow/python/data/util/nest.py", line 36, in <module>
>     from tensorflow.python.framework import sparse_tensor as _sparse_tensor
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/tensorflow/python/framework/sparse_tensor.py", line 24, in <module>
>     from tensorflow.python.framework import constant_op
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/tensorflow/python/framework/constant_op.py", line 25, in <module>
>     from tensorflow.python.eager import execute
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/tensorflow/python/eager/execute.py", line 23, in <module>
>     from tensorflow.python.framework import dtypes
>   File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/tensorflow/python/framework/dtypes.py", line 29, in <module>
>     _np_bfloat16 = _pywrap_bfloat16.TF_bfloat16_type()
> TypeError: Unable to convert function return value to a Python type! The signature was
> 	() -> handle

I have tried different versions of numpy which support python 3.10 neither of them works with spacy. Looks like a bug.

## How to reproduce the behaviour
`pip install 'spacy[apple]'`

## Your Environment
* Operating System: Mac OS Monterey 12.5.1
* Python Version Used: 3.10.4
* spaCy Version Used: 3.2, 3.4.1
* Environment Information: Numpy 1.22, 1.21.4, 1.21.5 (neither of them work)

>   - catalogue [required: >=2.0.6,<2.1.0, installed: 2.0.8]
>   - cymem [required: >=2.0.2,<2.1.0, installed: 2.0.6]
>   - jinja2 [required: Any, installed: 3.0.3]
>     - MarkupSafe [required: >=2.0, installed: 2.1.1]
>   - langcodes [required: >=3.2.0,<4.0.0, installed: 3.3.0]
>   - murmurhash [required: >=0.28.0,<1.1.0, installed: 1.0.8]
>   - numpy [required: >=1.15.0, installed: 1.21.4]
>   - packaging [required: >=20.0, installed: 21.3]
>     - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
>   - pathy [required: >=0.3.5, installed: 0.6.2]
>     - smart-open [required: >=5.2.1,<6.0.0, installed: 5.2.1]
>     - typer [required: >=0.3.0,<1.0.0, installed: 0.4.2]
>       - click [required: >=7.1.1,<9.0.0, installed: 8.1.3]
>   - preshed [required: >=3.0.2,<3.1.0, installed: 3.0.7]
>     - cymem [required: >=2.0.2,<2.1.0, installed: 2.0.6]
>     - murmurhash [required: >=0.28.0,<1.1.0, installed: 1.0.8]
>   - pydantic [required: >=1.7.4,<1.10.0,!=1.8.1,!=1.8, installed: 1.8.2]
>     - typing-extensions [required: >=3.7.4.3, installed: 4.3.0]
>   - requests [required: >=2.13.0,<3.0.0, installed: 2.28.1]
>     - certifi [required: >=2017.4.17, installed: 2022.6.15]
>     - charset-normalizer [required: >=2,<3, installed: 2.0.4]
>     - idna [required: >=2.5,<4, installed: 3.3]
>     - urllib3 [required: >=1.21.1,<1.27, installed: 1.26.11]
>   - setuptools [required: Any, installed: 65.3.0]
>   - spacy-legacy [required: >=3.0.9,<3.1.0, installed: 3.0.10]
>   - spacy-loggers [required: >=1.0.0,<2.0.0, installed: 1.0.3]
>     - wasabi [required: >=0.8.1,<1.1.0, installed: 0.10.1]
>   - srsly [required: >=2.4.3,<3.0.0, installed: 2.4.4]
>     - catalogue [required: >=2.0.3,<2.1.0, installed: 2.0.8]
>   - thinc [required: >=8.1.0,<8.2.0, installed: 8.1.1]
>     - blis [required: >=0.7.8,<0.10.0, installed: 0.7.8]
>       - numpy [required: >=1.15.0, installed: 1.21.4]
>     - catalogue [required: >=2.0.4,<2.1.0, installed: 2.0.8]
>     - confection [required: >=0.0.1,<1.0.0, installed: 0.0.1]
>       - pydantic [required: >=1.7.4,<1.10.0,!=1.8.1,!=1.8, installed: 1.8.2]
>         - typing-extensions [required: >=3.7.4.3, installed: 4.3.0]
>       - srsly [required: >=2.4.0,<3.0.0, installed: 2.4.4]
>         - catalogue [required: >=2.0.3,<2.1.0, installed: 2.0.8]
>     - cymem [required: >=2.0.2,<2.1.0, installed: 2.0.6]
>     - murmurhash [required: >=1.0.2,<1.1.0, installed: 1.0.8]
>     - numpy [required: >=1.15.0, installed: 1.21.4]
>     - preshed [required: >=3.0.2,<3.1.0, installed: 3.0.7]
>       - cymem [required: >=2.0.2,<2.1.0, installed: 2.0.6]
>       - murmurhash [required: >=0.28.0,<1.1.0, installed: 1.0.8]
>     - pydantic [required: >=1.7.4,<1.10.0,!=1.8.1,!=1.8, installed: 1.8.2]
>       - typing-extensions [required: >=3.7.4.3, installed: 4.3.0]
>     - setuptools [required: Any, installed: 65.3.0]
>     - srsly [required: >=2.4.0,<3.0.0, installed: 2.4.4]
>       - catalogue [required: >=2.0.3,<2.1.0, installed: 2.0.8]
>     - wasabi [required: >=0.8.1,<1.1.0, installed: 0.10.1]
>   - tqdm [required: >=4.38.0,<5.0.0, installed: 4.64.1]
>   - typer [required: >=0.3.0,<0.5.0, installed: 0.4.2]
>     - click [required: >=7.1.1,<9.0.0, installed: 8.1.3]
>   - wasabi [required: >=0.9.1,<1.1.0, installed: 0.10.1]

**Comment by adrianeboyd** [comment]
Hmm, that looks like it's related to `tensorflow`. Does the spacy install work correctly in an environment where tensorflow is **not** installed?

**Comment by alankabisov** [comment, marked as answer]
I've removed Tensorflow, now it shows following error:
```
Traceback (most recent call last):
  File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/runpy.py", line 187, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/runpy.py", line 146, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/runpy.py", line 110, in _get_module_details
    __import__(pkg_name)
  File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/spacy/__init__.py", line 6, in <module>
    from .errors import setup_default_warnings
  File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/spacy/errors.py", line 2, in <module>
    from .compat import Literal
  File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/spacy/compat.py", line 38, in <module>
    from thinc.api import Optimizer  # noqa: F401
  File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/thinc/api.py", line 2, in <module>
    from .initializers import normal_init, uniform_init, glorot_uniform_init, zero_init
  File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/thinc/initializers.py", line 4, in <module>
    from .backends import Ops
  File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/thinc/backends/__init__.py", line 7, in <module>
    from .ops import Ops
  File "/Users/alan/opt/anaconda3/envs/ds/lib/python3.10/site-packages/thinc/backends/ops.py", line 16, in <module>
    from .cblas import CBlas
  File "thinc/backends/cblas.pyx", line 1, in init thinc.backends.cblas
ImportError: blis.cy does not export expected C function sgemm
```

**Reply by adrianeboyd** [reply]
Can you start again in a new environment? I think there are conflicting package versions from various install attempts.

**Reply by alankabisov** [reply]
Hm.. On fresh environment it installs and load everything well.. Thank you so much! At least now I can run it on my machine. But how can I use it with Tensorflow or other packages?

**Reply by adrianeboyd** [reply]
I'd look into advice related to those tensorflow errors, one thread I found is: https://developer.apple.com/forums/thread/702851

**Reply by alankabisov** [reply]
Nice, thank you! I now installed packages versions as on fresh environment and reinstalled numpy and spacy works with other packages! Thank you so much for your help and guidance!:)
